app/main/helpers/frameworks/validation.py

```diff
@@ -0,0 +1,178 @@
+        raw_errors_map = self.errors()
+        errors_map = list()
+        for question_number, question_id in enumerate(self.all_fields()):
+            if question_id in raw_errors_map:
+                validation_message = self.get_error_message(question_id, raw_errors_map[question_id])
+                errors_map.append((question_id, {
+                    'input_name': question_id,
+                    'question': "Question {}".format(question_number + 1),
+                    'message': validation_message,
+                }))
+
+        return errors_map
+
+    def get_error_message(self, question_id, message_key):
+        for validation in self.content.get_question(question_id).get('validations', []):
+            if validation['name'] == message_key:
+                return validation['message']
+        default_messages = {
+            'answer_required': 'Answer required'
+        }
+        return default_messages.get(
+            message_key, 'There was a problem with the answer to this question')
+
+    def errors(self):
+        return {}
+
+    def all_fields(self):
+        return reduce(add, (section.get_question_ids() for section in self.content))
+
+    def fields_with_values(self):
+        # empty list types will not be included so only string types need to be considered
+        return set(key for key, value in self.answers.items()
+                   if not isinstance(value, six.string_types) or len(value) > 0)
+
+    def character_limit_errors(self):
+        errors_map = {}
+        for question_id in self.all_fields():
+            if self.content.get_question(question_id).get('type') in ['text', 'textbox_large']:
+                answer_length = len(self.answers.get(question_id, ""))
+                if (self.character_limit is not None and answer_length > self.character_limit):
+                    errors_map[question_id] = "under_character_limit"
+
+        return errors_map
+
+    def formatting_errors(self, answers):
+        errors_map = {}
+        if self.email_validation_fields is not None and len(self.email_validation_fields) > 0:
+            for field in self.email_validation_fields:
+                if not re.match(EMAIL_REGEX, self.answers.get(field, '')):
+                    errors_map[field] = 'invalid_format'
+        return errors_map
+
+
+class EmptyValidator(FrameworkValidator):
+    """
+    An empty validator that will always pass.
+    """
+    pass
+
+
+class G7Validator(FrameworkValidator):
+    """
+    Validator for G-Cloud 7.
+    """
+    optional_fields = G7_OPTIONAL_FIELDS
+    email_validation_fields = G7_EMAIL_VALIDATION_FIELDS
+    character_limit = G7_CHARACTER_LIMIT
+
+    def errors(self):
+        errors_map = {}
+        errors_map.update(self.character_limit_errors())
+        errors_map.update(self.formatting_errors(self.answers))
+        errors_map.update(self.answer_required_errors())
+        return errors_map
+
+    def answer_required_errors(self):
```

Comment:
I think this could be easily moved to the parent class (were the rest of the `_errors` methods are defined) if we make `required_fields` a property.